### ISSUE TITLE
`tuist cache` failed for Firebase SDK 10.24.0

### ISSUE THREAD

**Issue by kapitoshka438** [tuist/tuist#6249]
### What happened?

After upgrading Tuist to 4.11.0 caching Firebase SDK stopped working.
It works fine with Tuist 4.9.0.

### How do we reproduce it?

Add Firebase SDK 10.24.0 as a dependency in Tuist/Package.swift.
Add these to the Target dependencies:
```
.external(name: "FirebaseAnalyticsSwift"),
.external(name: "FirebaseCrashlytics"),
.external(name: "FirebaseMessaging"),
.external(name: "FirebaseRemoteConfigSwift"),
```

### Error log

Mac OS 13.6.1, Xcode 15.2, Tuist 4.11.0
```
Build Succeeded
Creating XCFrameworks
Creating XCFramework for GoogleDataTransport
[x] error: cannot compute path of binary 'Path(str: "/private/var/folders/bn/w8flw_m92r3gty0pdt7kqly40000gq/T/TemporaryDirectory.bE5zeP/artifacts/iOS/device/GoogleDataTransport.framework/GoogleDataTransport")' relative to that of '/var/folders/bn/w8flw_m92r3gty0pdt7kqly40000gq/T/TemporaryDirectory.bE5zeP/artifacts/iOS/device/GoogleDataTransport.framework'
The 'xcodebuild' command exited with error code 70 and message:
error: cannot compute path of binary 'Path(str: "/private/var/folders/bn/w8flw_m92r3gty0pdt7kqly40000gq/T/TemporaryDirectory.bE5zeP/artifacts/iOS/device/GoogleDataTransport.framework/GoogleDataTransport")' relative to that of '/var/folders/bn/w8flw_m92r3gty0pdt7kqly40000gq/T/TemporaryDirectory.bE5zeP/artifacts/iOS/device/GoogleDataTransport.framework'
Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
```

Mac OS 13.6.6, Xcode 15.2, Tuist 4.11.0
```
Build Succeeded
Creating XCFrameworks
Creating XCFramework for GoogleUtilities-AppDelegateSwizzler
[x] error: cannot compute path of binary 'Path(str: "/private/var/folders/yx/8j8fh09j2nbbyz1gslv1m88m0000gp/T/TemporaryDirectory.FVbQvI/artifacts/iOS/simulator/GoogleUtilities_AppDelegateSwizzler.framework/GoogleUtilities_AppDelegateSwizzler")' relative to that of '/var/folders/yx/8j8fh09j2nbbyz1gslv1m88m0000gp/T/TemporaryDirectory.FVbQvI/artifacts/iOS/simulator/GoogleUtilities_AppDelegateSwizzler.framework'
The 'xcodebuild' command exited with error code 70 and message:
error: cannot compute path of binary 'Path(str: "/private/var/folders/yx/8j8fh09j2nbbyz1gslv1m88m0000gp/T/TemporaryDirectory.FVbQvI/artifacts/iOS/simulator/GoogleUtilities_AppDelegateSwizzler.framework/GoogleUtilities_AppDelegateSwizzler")' relative to that of '/var/folders/yx/8j8fh09j2nbbyz1gslv1m88m0000gp/T/TemporaryDirectory.FVbQvI/artifacts/iOS/simulator/GoogleUtilities_AppDelegateSwizzler.framework'
Consider creating an issue using the following link: https://github.com/tuist/tuist/issues/new/choose
```

### macOS version

13.6.1, 13.6.6

### Tuist version

4.11.0

### Xcode version

15.2

**Comment by kapitoshka438** [comment]
Seems like this is not related to Firebase. Below is the crash log for Tuist 4.11.0 + Xcode 15.2. It's just failing on the first dependency to create an xcframework:
```
Build Succeeded
Creating XCFrameworks
Creating XCFramework for AppMetricaEncodingUtils
❌ error: cannot compute path of binary 'Path(str: "/private/var/folders/7s/1pbc1g_17pd13t4dpvmjh7wr0000gp/T/TemporaryDirectory.7oIkE9/artifacts/iOS/device/AppMetricaEncodingUtils.framework/AppMetricaEncodingUtils")' relative to that of '/var/folders/7s/1pbc1g_17pd13t4dpvmjh7wr0000gp/T/TemporaryDirectory.7oIkE9/artifacts/iOS/device/AppMetricaEncodingUtils.framework'
```

Tuist 4.9.0 + Xcode 15.2 works fine, as well as Tuist 4.11.0 + Xcode 15.3

**Comment by pepicrft** [comment]
It's a bug in Xcode 15.2 when the paths contain symlinks. If you can't update to Xcode 15.3, we'll be rolling out a fix in the next version of Tuist.

**Comment by kapitoshka438** [comment]
Fixed in 4.13.0 thank you @pepicrft
